src/rand/rand.zig

```diff
@@ -426,3 +426,75 @@ const rust_expected_weights: [1000]u64 = .{
     40566523540155677,  171562338978082216, 151758301674069632, 173398251098334249, 63757156897079302,
     81032160488073578,  173298561206951917, 154421360720272045, 15981567533895039,  88393403508710277,
 };
+
+/// Implementation of Alastair J. Walker's "Alias method".
+/// Once constructed, this allows for efficiently getting pseudorandom indeces
+/// that fit the given probabilities.
+pub const WeightedAliasSampler = struct {
+    probability: []const f32,
+    alias: []const usize,
+    pub fn init(allocator: std.mem.Allocator, probabilities: []const f32) !WeightedAliasSampler {
```

Comment:
The sampler will silently misbehave if the probabilities do not add up to 1.

Either of these changes will make the sampler safer:
1. return error if they do not sum to 1
2. treat them as weights instead of probabilities

Personally I prefer 2 because it's more flexible, won't result in runtime errors, and it makes the api more consistent with the other sampler. What do you think?

Reply:
https://github.com/Syndica/sig/pull/325/commits/8adc271af982bddc22bacde3020a617314895ae5